crates/junction-core/src/client.rs

```diff
@@ -194,6 +194,35 @@ impl Client {
         backends
     }
 
+    /// Resolve an HTTP method, URL, and headers to a target backend, returning
+    /// the Route that matched, the index of the rule that matched, and the
+    /// backend that the [Target] that was selected.
+    ///
+    /// This is equivalent to calling [check_route](crate::check_route) with the
+    /// result of [Self::dump_routes] but will use the client's live dynamic
+    /// configuration and defaults instead of a static snapshot.
+    ///
+    /// Checking a route will not dynamically fetch configuration for a route
+    /// that has not been requested with
+    /// [resolve_http](Self::resolve_http).
+    pub fn check_route(
+        &self,
+        method: &http::Method,
+        url: &crate::Url,
+        headers: &http::HeaderMap,
+    ) -> crate::Result<(Arc<Route>, usize, Target)> {
+        let request = HttpRequest {
+            method,
+            url,
+            headers,
+        };
+
+        let targets = targets_for_url(url)?;
```

Comment:
If we do want to let this modify configuration, it'd be fairly trivial to also subscribe to all targets here.